app/models/cdr/cdr.rb

```diff
@@ -325,6 +325,16 @@ def self.pending_events
     SqlCaller::Cdr.select_value("select pending_events from pgq.get_consumer_info('cdr_billing', 'cdr_billing')")
   end
 
+  scope :routing_tag_ids_array_contains, ->(*tag_id) { where.contains routing_tag_ids: Array(tag_id) }
+
+  scope :tagged, lambda { |value|
+    if ActiveModel::Type::Boolean.new.cast(value)
+      where("routing_tag_ids <> '{}'") # has tags
+    else
+      where("routing_tag_ids = '{}' OR routing_tag_ids IS NULL") # no tags
+    end
+  }
+
```

Comment:
this commit should not be in MR

you need to rebase on master instead of adding this code here

Reply:
looks like you've mix files from previous MR

you need to rebase it against master using

```bash
git pull --rebase upstream master
```

resolve conflicts if needed,

and after that remove extra commits from this PR if needed via `git rebase -i ...`